arangod/Aql/AstNode.cpp

```diff
@@ -2860,6 +2860,15 @@ void AstNode::setStringValue(char const* v, size_t length) {
   value.length = static_cast<uint32_t>(length);
 }
 
+void AstNode::setStringValue(std::string const& v) {
+  auto length = size_t{v.length()};
+  auto value = std::make_unique<char[]>(length);
+
+  std::memcpy(value.get(), v.c_str(), length);
+
+  setStringValue(value.release(), length);
+}
```

Comment:
No, please don't do this.
It completely bypasses the existing memory management of AQL.